### PR TITLE
Use memmap2 for mmap logic

### PR DESCRIPTION
**Pull request by norcalli**
As discussed in #1 

Also adds support for remap on non-linux OSes

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -26,11 +22,30 @@ pub struct MmapAppend {
...
+}
+
+#[cfg(target_os = "linux")]
+fn remap(_fd: RawFd, inner: &mut MmapRaw, new_len: usize) -> Result<()> {
+    unsafe { inner.remap(new_len, memmap2::RemapOptions::new().may_move(true)) }
+}
+
+#[cfg(not(target_os = "linux"))]
+fn remap(fd: RawFd, inner: &mut MmapRaw, new_len: usize) -> Result<()> {
+    inner.flush()?;
```
Comment (norcalli) [inline src/lib.rs:40]: I'm not 100% if the flush is necessary.

**Comment by mikedilger** [comment]
I'm not confident that I can even evaluate this PR at this point. My understanding of this crate (even though I wrote it) has faded from my memory.

Would you be interested in taking it over?

**Comment by norcalli** [comment]
Understood. Yeah I can take over. My crates username is “norcalli” since this is the same linked account.

**Comment by mikedilger** [comment]
Ok I added you as owner.  You can remove me if you want.  Just clone this repo elsewhere and change the crate to point to your clone. Then I'll archive this one.

**Pull request merged**